src/OrchardCore/OrchardCore.Setup.Abstractions/Events/ISetupSuccessEventHandler.cs

```diff
@@ -0,0 +1,15 @@
+using System.Threading.Tasks;
+
+namespace OrchardCore.Setup.Events
+{
+    /// <summary>
+    /// Setup success event handler.
+    /// </summary>
+    public interface ISetupSuccessEventHandler
```

Comment:
Maybe `IShellSetupEvent`

Reply:
Yes or maybe `ITenantSetupEvent` or `ITenantSetupHandler` or just `ISetupHandler`.

But not sure it makes a good difference from the existing `ISetupEventHandler`.

Hmm or maybe `ISetupSuccessHandler` because `Success` is an event.

Or add the new method to the existing `ISetupEventHandler`, anyway we should rename the existing `Setup()` method to `SetupAsync()`, but then need to use the obsolete attribute and so on.

Or we could add a `SetupAsync()` to `IModularTenantEvents`, less breaking as if someone has a class deriving from the `ModularTenantEvents` base class it will still compile, but then more changes.

For now I kept the separate handler and named it `ISetupSuccessHandler`.

Reply:
@jtkech 
I actually do no like using `Success` in the interface name. That would be part of the method that corresponds to the event. Here is what I think is a better approach while keeping the code backward compatible. 

Mark the existing `ISetupEventHandler` interface as obsolete. But continue to call it for backward compatibility. Create a new interface `ITenantSetupHandler` or `IShellSetupHandler` this new interface will replace the old `ISetupEventHandler` and will look something like this

```C#
public interface ITenantSetupHandler
{
    Task SettingUpAsync(SetupContext context);

    Task FinishedAsync(CompletedSetupContext context);
}

public class CompletedSetupContext 
{
    public bool Success { get; set; }

    public string[] Errors { get; set; }
}
```
Also, I would not call the `FinishedAsync()` using a background job. We may want to perform action immediately when the setup finished during the same request like display notification on the screen.

If someone want to run a background job, then can implement the new `ITenantSetupHandler` and execute logic in the background.

Reply:
Yes, was too lazy to update the interfaces, only wanted to do an additive change without changing the existing interfaces, I will think about it.

> We may want to perform action immediately when the setup finished during the same request like display notification on the screen.

As I can see this is what the current `ISetupEventHandler` do by providing errors that the setup controller will notify.

> If someone want to run a background job, then can implement the new ITenantSetupHandler and execute logic in the background.

So nothing to do as it is already possible through the existing `ISetupEventHandler`, just wanted to add it oob.

Reply:
Don't like the name `Finished`, I prefer for example `Completed`, maybe no problem to have `Success` in the name of an additive handler that handles only one event.

But never fully sure ;)

Reply:
For example we have `IRoleRemovedEventHandler` and `IRoleCreatedEventHandler`, yes maybe better to have only one `IRoleEventHandler` as we have for `IUserEventHandler`.

But when an handler dedicated to only one event already exists, I'm fine to use the same way to handle a new event, at least for a given event it allows to only resolve the related handlers.

Reply:
For the setup success event, triggered after the request, I didn't pass a context because for example you can just inject `ShellSettings` to know the related tenant.

But yes, if we want to handle more events, methods can be grouped in the same handler interface and with the needed contexts.

Reply:
@jtkech  
> IRoleRemovedEventHandler and IRoleCreatedEventHandler, yes maybe better to have only one IRoleEventHandler as we have for IUserEventHandler.

I still feel that my suggested approach is cleaner. With the current approach you are taking in this PR, you are following the problem we have in Roles. My suggested approach will prevent similar problem. But, you know what is best.

Also, I suggested not to call the success from a background to allow us to do same-request logic. Thoughts?

Reply:
Okay, anyway we are too close to the next release, and here it was more to show you a way to fix your issue. So I think I will close this PR, it will be easy to suggest another one after the incoming release and discuss again about refactoring handler interfaces.